program/src/state.rs

```diff
@@ -245,6 +248,48 @@ impl IsInitialized for Oracle {
 }
 impl InitBorshState for Oracle {}
 
+/// Requester data.
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, BorshSchema, Default, PartialEq)]
+pub struct Requester {
+    /// is usually the oracle name
+    pub description: [u8; 32],
+    /// is initialized
+    pub is_initialized: bool,
+    /// TODO: do we need this withdrawable
+    pub withdrawable: u64,
```

Comment:
remove withdrawable.

there is a subtle security issue where Requester & Oracle are binary compatible, and allowing either roles to act as the other.

removing `withdrawable` so Requester & Oracle are not binary compatible.